FrameworkSystem/DB/ProxyDB.py

```diff
@@ -624,22 +765,30 @@ def getProxy(self, userDN, userGroup, requiredLifeTime=False):
     # Standard proxy is requested
     retVal = self.__getPemAndTimeLeft(userDN, userGroup)
     if not retVal['OK']:
-      return retVal
-    pemData = retVal['Value'][0]
-    timeLeft = retVal['Value'][1]
-    chain = X509Chain()
-    retVal = chain.loadProxyFromString(pemData)
-    if not retVal['OK']:
-      return retVal
-    if requiredLifeTime:
-      if timeLeft < requiredLifeTime:
-        if self.__useMyProxy:
-          retVal = self.renewFromMyProxy(userDN, userGroup, lifeTime=requiredLifeTime, chain=chain)
-          if not retVal['OK']:
-            return S_ERROR("Can't get a proxy for %s seconds: %s" % (requiredLifeTime, retVal['Message']))
-          chain = retVal['Value']
-        else:
-          return S_ERROR("Can't get a proxy: the required lifetime is less than the time left in the proxy")
+      errMsg = retVal['Message']
+      retVal = self.__generateProxyFromProxyProviders(userDN, userGroup, requiredLifeTime=requiredLifeTime)
+    elif requiredLifeTime:
+      if retVal['Value'][1] < requiredLifeTime and not self.__useMyProxy:
+        errMsg = 'the required lifetime is less than the time left in the proxy'
+        retVal = self.__generateProxyFromProxyProviders(userDN, userGroup, requiredLifeTime=requiredLifeTime)
+    if retVal['OK']:
```

Comment:
If you do
```
if not retVal['OK']:
  return retVal
```
You can save on one indentation level in the next block

Reply:
fixed